src/usr/local/www/diag_packet_capture.php

```diff
@@ -101,6 +101,13 @@ function fixup_host($value, $position) {
 	}
 }
 
+function fixup_port($value, $position) {
```

Comment:
There is a lot of repeated code here that could easily be avoided. Make the `fixup_host()` function generic with an extra parameter to set the type (`host` or `port`) and adjust tests/output as needed to avoid all the copy/pasted code with small changes.

Reply:
all repeated code consolidated to generic functions:
fixup_hostport
precheck_hostport
hostport_array_fixer